cmd/oras/root/tag.go

```diff
@@ -74,11 +74,16 @@ Example - Tag the manifest 'v1.0.1' to 'v1.0.2' in an OCI image layout folder 'l
 		},
 		PreRunE: func(cmd *cobra.Command, args []string) error {
 			opts.RawReference = args[0]
-			if _, err := registry.ParseReference(opts.RawReference); err != nil {
-				return fmt.Errorf("unable to add tag for '%s': %w", opts.RawReference, err)
-			}
 			opts.targetRefs = args[1:]
-			return option.Parse(&opts)
+			if err := option.Parse(&opts); err != nil {
+				if inner, ok := err.(*oerrors.Error); ok {
```

Comment:
I'm curious about the reason why this type assertion is required?

Reply:
`errors.Is()` does `Unwarp` the inner errors.

See https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/errors/wrap.go;l=44-78

Reply:
I see the `inner` is used later.